sgv2-dynamo-api/src/main/java/io/stargate/sgv2/dynamosvc/dynamo/QueryProxy.java

```diff
@@ -0,0 +1,225 @@
+package io.stargate.sgv2.dynamosvc.dynamo;
+
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.QueryRequest;
+import com.amazonaws.services.dynamodbv2.model.QueryResult;
+import io.stargate.proto.QueryOuterClass;
+import io.stargate.proto.Schema;
+import io.stargate.sgv2.common.cql.builder.Predicate;
+import io.stargate.sgv2.common.cql.builder.QueryBuilder;
+import io.stargate.sgv2.common.grpc.StargateBridgeClient;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class QueryProxy extends Proxy {
+
+  private static final Logger logger = LoggerFactory.getLogger(QueryProxy.class);
+
+  private static final Map<String, Predicate> operatorConverter =
+      new HashMap<String, Predicate>() {
+        {
+          put("=", Predicate.EQ);
+          put("<=", Predicate.LTE);
+          put("<", Predicate.LT);
+          put(">=", Predicate.GTE);
+          put(">", Predicate.GT);
+        }
+      };
+
+  /**
+   * Basic partition key equality pattern
+   *
+   * <p>Example:<br>
+   * partitionKeyName = :partitionkeyval
+   */
+  private static final Pattern EQUALITY_PATTERN =
+      Pattern.compile("^\\s*(\\S+)\\s*=\\s*(:\\S+)\\s*$");
+
+  /**
+   * Basic partition key equality pattern + Sort key operation comparison pattern
+   *
+   * <p>Examples:<br>
+   * partitionKeyName = :partitionkeyval AND sortKeyName = :sortkeyval<br>
+   * partitionKeyName = :partitionkeyval AND sortKeyName < :sortkeyval<br>
+   * partitionKeyName = :partitionkeyval AND sortKeyName <= :sortkeyval<br>
+   * partitionKeyName = :partitionkeyval AND sortKeyName > :sortkeyval<br>
+   * partitionKeyName = :partitionkeyval AND sortKeyName >= :sortkeyval<br>
+   */
+  private static final Pattern OP_COMPARISON_PATTERN =
+      Pattern.compile("(\\S+)\\s*=\\s*(:\\S+)\\s*AND\\s*(\\S+)\\s*([<>=]+)\\s*(:\\S+)");
```

Comment:
We should split this pattern into two parts. First, we match the equality pattern, and then we match the operator comparison pattern for the unmatched substring. By doing so, we could reuse the regex and match logic in FilterExpression.